itou/geo/utils.py

```diff
@@ -11,3 +19,35 @@ def coords_to_geometry(lat, lon) -> Optional[GEOSGeometry]:
 
 def multipolygon_to_geometry(spec: str) -> GEOSGeometry:
     return GEOSGeometry(f"{spec}")
+
+
+def geographic_eligibility_criteria(job_seeker) -> Dict:
+    """Get geographic eligibility criteria for given job seeker:
+        - QPV
+        - ZRR (partial or complete)
+    Used as info / reminder in eligibility forms for SIAE / prescriber organization.
+    """
+
+    known_criteria = {}
+
+    try:
+        if not QPV.in_qpv(job_seeker, geom_field="coords"):
+            known_criteria |= {"address_in_qpv": job_seeker.address_on_one_line}
+    except ValueError as ex:
+        logger.warning(f"Can't get geolocation of user pk={job_seeker.pk}: {ex}")
+
+    try:
+        # There's currently no direct link between User.city and Cities.City
+        job_seeker_city = City.objects.get(slug=job_seeker.city_slug)
+    except City.DoesNotExist:
+        logger.warning(
+            f"Can't match INSEE code for user pk={job_seeker.pk}: "
+            f"city '{job_seeker.city}' has no match in 'Cities.City' model"
+        )
+    else:
+        if not ZRR.in_zrr(job_seeker_city.code_insee):
+            known_criteria |= {"commune_in_zrr": job_seeker_city.name}
+        elif ZRR.partially_in_zrr(job_seeker_city.code_insee):
+            known_criteria |= {"commune_partially_in_zrr": job_seeker_city.name}
+
+    return known_criteria
```

Comment:
Je trouve cette logique dommage, car cela te force à avoir un dictionnaire sur lequel on va devoir refaire des itérations, des ifs, etc. et ça mélange deux concepts proches mais pas totalement identiques.

Par ailleurs, on passe par une classmethod de QPV et ZRR pour finalement retomber sur un `job_seeker` à chaque fois.

Je te conseille une autre approche (je sais, ça va faire un peu reprendre le code de cette PR, mais je pense que c'est pour le mieux)

Sur le modèle User, ajouter 2 `@cached_property`:
- `qpv_address` qui retourne l'adresse en QPV ou None
- `zrr_city_name` qui retourne un tuple `(city_name, is_partially_zrr)` ou `None` suivant que c'est partiel ou non

Ca sera plus approprié parce que:
- pas besoin d'en faire des caisses avec plein de methodes et classmethods qui en fait sont juste des trucs au-dessus de filter
- on comprend bien que etre partially en ZRR ou non est bien antinomique, c'est l'un ou l'autre
- on peut etre en QPV et en ZRR
- c'est utilisable directement dans tes templates sans dict: `{% if user.qpv_address or user.zrr_city_name %}` ...
- pas besoin d'injecter un nouvel objet de contexte

Après on peut garder cette version si ma proposition t'ennuie mais je la trouve moins testable et avec vraiment plus d'indirections.

Reply:
Non, ça ne m'ennuie pas :) C'est argumenté, et j'ai déjà supprimé les wrappers de ZRR.